File: ueueo-framework/ueueo-localization/src/main/java/Abp/Localization/VirtualFiles/Json/JsonVirtualFileLocalizationResourceContributor.cs
using Microsoft.Extensions.FileProviders;
using System;
using Volo.Abp.Localization.Json;

namespace Volo.Abp.Localization.VirtualFiles.Json;

//TODO: Use composition over inheritance..?

public class JsonVirtualFileLocalizationResourceContributor : VirtualFileLocalizationResourceContributorBase
{
    public JsonVirtualFileLocalizationResourceContributor(String virtualPath)
        : base(virtualPath)
    {

    }

    protected override boolean CanParseFile(IFileInfo file)
    {
        return file.Name.EndsWith(".json", StringComparison.OrdinalIgnoreCase);
    }

    protected override ILocalizationDictionary CreateDictionaryFromFileContent(String jsonString)
    {
        return JsonLocalizationDictionaryBuilder.BuildFromJsonString(jsonString);
    }
}

File: ueueo-framework/ueueo-localization/src/main/java/com/ueueo/localization/virtualfiles/json/JsonVirtualFileLocalizationResourceContributor.java
package com.ueueo.localization.virtualfiles.json;

//TODO: Use composition over inheritance..?

import com.ueueo.localization.ILocalizationDictionary;
import com.ueueo.localization.json.JsonLocalizationDictionaryBuilder;
import com.ueueo.localization.virtualfiles.VirtualFileLocalizationResourceContributorBase;
import org.apache.commons.lang3.StringUtils;

import java.io.File;

public class JsonVirtualFileLocalizationResourceContributor extends VirtualFileLocalizationResourceContributorBase {
    public JsonVirtualFileLocalizationResourceContributor(String virtualPath) {
        super(virtualPath);
    }

    @Override
    protected boolean canParseFile(File file) {
        return StringUtils.endsWithIgnoreCase(file.getName(), ".json");
    }

    @Override
    protected ILocalizationDictionary createDictionaryFromFileContent(String jsonString) {
        return JsonLocalizationDictionaryBuilder.buildFromJsonString(jsonString);
    }
}
